src/lib/gitutils.ts

```diff
@@ -0,0 +1,130 @@
+import GitUrlParse from "git-url-parse";
+import { logger } from "../logger";
+import { exec } from "./shell";
+
+/**
+ * Gets the current working branch.
+ */
+export const getCurrentBranch = async () => {
+  try {
+    const branch = await exec("git", ["rev-parse", "--abbrev-ref", "HEAD"]);
+    return branch;
+  } catch (_) {
+    logger.warn("Unable to determine current branch.");
+    return "";
+  }
+};
+
+/**
+ * Checkout the given branch.
+ *
+ * @param branchName
+ */
+export const checkoutBranch = async (branchName: string) => {
+  try {
+    await exec("git", ["checkout", `${branchName}`]);
+  } catch (_) {
+    logger.warn(`unable to checkout git branch ${branchName}.`);
```

Comment:
you may want to throw these errors instead of suppressing them, for debuggability. you should do this here, and everywhere else you're catching and logging warnings